roles/test-utils/sv1-mining-device/src/main.rs

```diff
@@ -1,9 +1,16 @@
 pub(crate) mod client;
 pub(crate) mod job;
 pub(crate) mod miner;
+use std::str::FromStr;
+
 pub(crate) use client::Client;
 
 #[async_std::main]
 async fn main() {
-    Client::connect(80).await
+    const ADDR: &str = "127.0.0.1:34255";
+    Client::connect(
+        80,
+        std::net::SocketAddr::from_str(ADDR).expect("Invalid upstream address"),
```

Comment:
```suggestion
       SocketAddr::from_str(ADDR).expect("Invalid upstream address"),
```
`std::net::SocketAddr` can be added to the import statement above.